libp2p/connmanager.nim

```diff
@@ -224,6 +225,32 @@ proc cleanupConn(c: ConnManager, conn: Connection) {.async.} =
 
   trace "Connection cleaned up", conn
 
+proc peerStartup(c: ConnManager, conn: Connection) {.async.} =
+  try:
+    trace "Triggering peer and connection events on connect", conn
+    let peerId = conn.peerInfo.peerId
+    await c.triggerPeerEvents(peerId, PeerEvent.Joined)
```

Comment:
this looks wrong - we should only trigger the Joined peer event on the _first_ connection to the peer, no? likewise for the `Left` - it would only happen when the last connection is fully cleaned up

Reply:
This is taken care of in the `triggerPeerEvents` call - this hasn't changed at all, just moved from the switch, we also test for this and I haven't yet seen this triggered more than once.